tests/TestCase/QueryTest.php

```diff
@@ -223,6 +224,70 @@ public function testJsonSerialize()
         $this->assertEquals(json_encode($expected), json_encode($this->query));
     }
 
+    public function testAndWhere()
+    {
+        $conditions = [
+            'foo' => 'bar',
+            'baz' => 2
+        ];
+        $this->query->andWhere($conditions);
+
+        $this->assertSame($conditions, $this->query->clause('where'));
+    }
+
+    public function testSelectWithArrayMerging()
+    {
+        $this->query->select(['id', 'name', 'title', 'description']);
+        $this->assertSame(['id', 'name', 'title', 'description'], $this->query->clause('select'));
+
+        $this->query->select(['published']);
+        $this->assertSame(['id', 'name', 'title', 'description', 'published'], $this->query->clause('select'));
+    }
+
+    public function testSelectWithArrayOverwrite()
+    {
+        $firstFields = ['id', 'first_name', 'last_name', 'date_of_birth'];
+        $this->query->select($firstFields);
+        $this->assertSame($firstFields, $this->query->clause('select'));
+
+        $secondFields = ['id', 'username', 'email'];
+        $this->query->select($secondFields, true);
+        $this->assertSame($secondFields, $this->query->clause('select'));
+    }
+
+    public function testSelectWithString()
+    {
+        $field = 'examples';
+        $this->query->select($field);
+
+        $this->assertSame([$field], $this->query->clause('select'));
+    }
+
+    public function testSelectWithExpression()
+    {
+        $exp = new Comparison('upvotes', 50, 'integer', '>=');
+        $this->query->select($exp);
+
+    /** @var Comparison $comparisonClause */
```

Comment:
I had to indent it like this to pass PHPCS. I think when we upgrade to the 3.x releases of the Code Sniffer we can fix this.